lib/active_fedora/identifiable.rb

```diff
@@ -83,6 +83,28 @@ def from_uri(uri, _)
       def abstract_class?
         self == Base
       end
+
+      # Returns the class descending directly from ActiveRecord::Base, or
+      # an abstract class, if any, in the inheritance hierarchy.
+      #
+      # If A extends ActiveRecord::Base, A.base_class will return A. If B descends from A
+      # through some arbitrarily deep hierarchy, B.base_class will return A.
+      #
+      # If B < A and C < B and if A is an abstract_class then both B.base_class
+      # and C.base_class would return B as the answer since A is an abstract_class.
+      def base_class
+        return File if self == File || superclass == File
```

Comment:
Something's not quite right about our class hierarchy, and there's also something funny about our associations that tries to get the base class for`AF::Base`.